api/src/test/java/org/pmiops/workbench/ras/RasLinkServiceTest.java

```diff
@@ -0,0 +1,167 @@
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Scope;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class RasLinkServiceTest {
+  private static final Timestamp NOW = Timestamp.from(Instant.now());
+  private static final FakeClock CLOCK = new FakeClock(NOW.toInstant(), ZoneId.systemDefault());
+
+  private static final String AUTH_CODE = "code";
+  private static final String REDIRECT_URL = "url";
+  private static final String ACCESS_TOKEN = "access_token_1";
+  private static final String LOGIN_GOV_USERNAME = "foo@gmail.com";
+  private static final String USER_INFO_JSON_LOGIN_GOV =
+      "{\"preferred_username\":\"user1@Login.Gov.com\", \"email\":\"" + LOGIN_GOV_USERNAME + "\"}";
+  private static final String USER_INFO_JSON_ERA =
+      "{\"preferred_username\":\"user2@eraCommons.com\", \"email\":\"" + LOGIN_GOV_USERNAME + "\"}";
+  private static final String ID_TOKEN_JWT_IAL_1 =
+      JWT.create()
+          .withClaim(
+              ACR_CLAIM,
+              "https://stsstg.nih.gov/assurance/ial/1 https://stsstg.nih.gov/assurance/aal/1")
+          .sign(Algorithm.none());
+  private static final String ID_TOKEN_JWT_IAL_2 =
+      JWT.create()
+          .withClaim(
+              ACR_CLAIM,
+              "https://stsstg.nih.gov/assurance/ial/2 https://stsstg.nih.gov/assurance/aal/1")
+          .sign(Algorithm.none());
+  private static final TokenResponse TOKEN_RESPONSE_IAL1 =
+      new TokenResponse().setAccessToken(ACCESS_TOKEN).set(Id_TOKEN_FIELD_NAME, ID_TOKEN_JWT_IAL_1);
+  private static final TokenResponse TOKEN_RESPONSE_IAL2 =
+      new TokenResponse().setAccessToken(ACCESS_TOKEN).set(Id_TOKEN_FIELD_NAME, ID_TOKEN_JWT_IAL_2);
+
+  private long userId;
+  private static DbUser currentUser;
+
+  private RasLinkService rasLinkService;
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  @Autowired private UserService userService;
+  @Autowired private UserDao userDao;
+  @Mock private OpenIdConnectClient mockOidcClient;
+
+  @TestConfiguration
+  @Import({
+    UserServiceTestConfiguration.class,
+  })
+  static class Configuration {
+    @Bean
+    Clock clock() {
+      return CLOCK;
+    }
+
+    @Bean
+    Random random() {
+      return new FakeLongRandom(123);
+    }
+
+    @Bean
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public WorkbenchConfig getWorkbenchConfig() {
+      return WorkbenchConfig.createEmptyConfig();
+    }
+
+    @Bean
+    @Scope("prototype")
+    DbUser user() {
+      return currentUser;
+    }
+  }
+
+  @MockBean UserServiceAuditor mockUserServiceAuditor;
+  @MockBean FireCloudService mockFireCloudService;
+  @MockBean ComplianceService mockComplianceService;
```

Comment:
Another way of delcaring these is on the test configuration, I don't remember whether this current approach you're using would allow injection into dependent services - the other approach may be required to make that work, but minimally this is more succinct syntax if you don't care about interacting with the mock at all. Example: https://github.com/all-of-us/workbench/blob/f56ff491ee1f303d434a89290e86dc944e6db0a4/api/src/test/java/org/pmiops/workbench/api/UserControllerTest.java#L72-L79